conans/client/manager.py

```diff
@@ -83,6 +83,7 @@ def deps_install(app, ref_or_path, install_folder, base_folder, graph_info, remo
         conanfile.folders.set_base_install(install_folder)
         conanfile.folders.set_base_imports(install_folder)
         conanfile.folders.set_base_generators(base_folder)
+        conanfile.folders.set_base_source(base_folder)
```

Comment:
This would also be assigned when `no layout defined`. Is that ok?

Reply:
I think so, some ``test_package`` like the reported one has:
```python
def generate(self):
    env = Environment()
    env.prepend_path("PKG_CONFIG_PATH", self.source_folder)
```
Will still want the ``self.source_folder`` as argument whether the ``layout()`` is defined or not. I was exactly trying to see if this could break something because of this.